src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  color: Color,
+  shape: Shape,
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(public color: Color, public a: number, public b: number,
+    public c: number) {
+    this.shape = 'triangle';
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `No side can be shorter than 0: a: ${a}, b: ${b}, c: ${c}`,
+      );
+    }
+
+    const sortedArray
+      = [this.a, this.b, this.c].sort((first, secound) => first - secound);
+    const max = sortedArray[sortedArray.length - 1];
+
+    switch (max) {
+      case c:
+        if (a + b <= max) {
+          throw new Error(
+            `The longest side of a triangle cannot be
+             shorter than the sum of the other two:
+             a: ${a}, b: ${b}, c: ${c}`,
+          );
+        }
+        break;
+      case a:
+        if (c + b <= max) {
+          throw new Error(
+            `The longest side of a triangle cannot be
+             shorter than the sum of the other two:
+             a: ${a}, b: ${b}, c: ${c}`,
+          );
+        }
+        break;
+
+      default:
+        if (a + c <= max) {
+          throw new Error(
+            `The longest side of a triangle cannot be
+             shorter than the sum of the other two:
+             a: ${a}, b: ${b}, c: ${c}`,
+          );
+        }
```

Comment:
please rewrite it so that there is no code duplication and unnecessary switch block